packages/next/server/base-server.ts

```diff
@@ -222,6 +200,63 @@ export default abstract class Server {
   protected abstract generatePublicRoutes(): Route[]
   protected abstract getFilesystemPaths(): Set<string>
 
+  protected abstract sendRenderResult(
+    req: BaseNextRequest,
+    res: BaseNextResponse,
+    options: {
+      result: RenderResult
+      type: 'html' | 'json'
+      generateEtags: boolean
+      poweredByHeader: boolean
+      options?: PayloadOptions
+    }
+  ): Promise<void>
+
+  protected abstract sendStatic(
+    req: BaseNextRequest,
+    res: BaseNextResponse,
+    path: string
+  ): Promise<void>
+
+  protected abstract runApi(
+    req: BaseNextRequest,
+    res: BaseNextResponse,
+    query: ParsedUrlQuery,
+    params: Params | boolean,
+    page: string,
+    builtPagePath: string
+  ): Promise<boolean>
+
+  protected abstract renderHTML(
+    req: BaseNextRequest,
+    res: BaseNextResponse,
+    pathname: string,
+    query: NextParsedUrlQuery,
+    renderOpts: RenderOpts
+  ): Promise<RenderResult | null>
+
+  protected abstract streamResponseChunk(
+    res: BaseNextResponse,
+    chunk: any
+  ): void
```

Comment:
Should this just go on the base response class? The web implementation will be a bit different as we'll need to get a writer and keep it open

Reply:
Yeah I think the abstraction here can be hidden in the response class